File: Command/CommandTextEditor/CommandTextEditor/AddTextCommand.cs
namespace CommandTextEditor
{
    public class AddTextCommand : ICommand//Concrete Command
    {
        private TextEditor _textEditor;
        private string _textToAdd;

        public AddTextCommand(TextEditor textEditor, string textToAdd)
        {
            _textEditor = textEditor;
            _textToAdd = textToAdd;
        }

        public void Execute() => _textEditor.InsertText(_textToAdd);
        public void Undo() => _textEditor.RemoveText(_textToAdd.Length);
    }
}

File: Command/JavaExample/CommandTextEditor/src/AddTextCommand.java
public class AddTextCommand implements ICommand {
    private TextEditor textEditor;
    private String textToAdd;

    public AddTextCommand(TextEditor textEditor, String textToAdd) {
        this.textEditor = textEditor;
        this.textToAdd = textToAdd;
    }

    @Override
    public void execute() {
        textEditor.insertText(textToAdd);
    }

    @Override
    public void undo() {
        textEditor.removeText(textToAdd.length());
    }
}